qiskit_nature/transformers/second_quantization/electronic/active_space_transformer.py

```diff
@@ -0,0 +1,391 @@
+                range(norbs_inactive, norbs_inactive + self._num_molecular_orbitals)
+            )
+        else:
+            active_orbs_idxs = self._active_orbitals
+            inactive_orbs_idxs = [
+                o
+                for o in range(nelec_total // 2)
+                if o not in self._active_orbitals and self._mo_occ_total[o] > 0
+            ]
+
+        return (active_orbs_idxs, inactive_orbs_idxs)
+
+    def _validate_num_electrons(self, nelec_inactive: int) -> None:
+        """Validates the number of electrons.
+
+        Args:
+            nelec_inactive: the computed number of inactive electrons.
+
+        Raises:
+            QiskitNatureError: if the number of inactive electrons is either negative or odd.
+        """
+        if nelec_inactive < 0:
+            raise QiskitNatureError("More electrons requested than available.")
+        if nelec_inactive % 2 != 0:
+            raise QiskitNatureError("The number of inactive electrons must be even.")
+
+    def _validate_num_orbitals(self, nelec_inactive: int, particle_number: ParticleNumber) -> None:
+        """Validates the number of orbitals.
+
+        Args:
+            nelec_inactive: the computed number of inactive electrons.
+            particle_number: the `ParticleNumber` containing system size information.
+
+        Raises:
+            QiskitNatureError: if more orbitals were requested than are available in total or if the
+                               number of selected orbitals mismatches the specified number of active
+                               orbitals.
+        """
+        if self._active_orbitals is None:
+            norbs_inactive = nelec_inactive // 2
+            if (
+                norbs_inactive + self._num_molecular_orbitals
+                > particle_number._num_spin_orbitals // 2
+            ):
+                raise QiskitNatureError("More orbitals requested than available.")
+        else:
+            if self._num_molecular_orbitals != len(self._active_orbitals):
+                raise QiskitNatureError(
+                    "The number of selected active orbital indices does not "
+                    "match the specified number of active orbitals."
+                )
+            if max(self._active_orbitals) >= particle_number._num_spin_orbitals // 2:
+                raise QiskitNatureError("More orbitals requested than available.")
+            if sum(self._mo_occ_total[self._active_orbitals]) != self._num_electrons:
+                raise QiskitNatureError(
+                    "The number of electrons in the selected active orbitals "
+                    "does not match the specified number of active electrons."
+                )
+
+    # TODO: can we efficiently extract this into the base class? At least the logic dealing with
+    # recursion is general and we should avoid having to duplicate it.
+    def _transform_property(self, prop: Property) -> Property:
+        """Transforms a Property object.
```

Comment:
Shall we address this TODO?

Reply:
Yes this is another open discussion point which I overlooked in my initial review. I think trying to avoid duplicating the property-type distinction in the future is a must. However, I would argue we postpone trying to abstract this away now. We don't have any other transformer ready now and I think it's probably easiest to do this once we actually see what another Transformer may look like.

Reply:
Alright let's leave this for the future then :+1: